From martyn.welch@gefanuc.com  Thu Oct 29 16:14:06 2009
From: Martyn Welch <martyn.welch@gefanuc.com>
Date: Thu, 29 Oct 2009 16:35:14 +0000
Subject: Staging: vme: Clean up tsi148 driver
To: gregkh@suse.de
Cc: devel@linuxdriverproject.org
Message-ID: <20091029163514.18368.56842.stgit@ES-J7S4D2J.amer.consind.ge.com>


* Remove message from IACK interrupt handler
* Correct clearing of location monitor interrupts
* Remove interrupt cleanup code that's duplcated in sub function

Signed-off-by: Martyn Welch <martyn.welch@gefanuc.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/staging/vme/bridges/vme_tsi148.c |   10 +---------
 1 file changed, 1 insertion(+), 9 deletions(-)

--- a/drivers/staging/vme/bridges/vme_tsi148.c
+++ b/drivers/staging/vme/bridges/vme_tsi148.c
@@ -237,7 +237,6 @@ static u32 tsi148_VERR_irqhandler(void)
  */
 static u32 tsi148_IACK_irqhandler(void)
 {
-	printk("tsi148_IACK_irqhandler\n");
 	wake_up(&iack_queue);
 
 	return TSI148_LCSR_INTC_IACKC;
@@ -2121,7 +2120,7 @@ int tsi148_lm_detach(struct vme_lm_resou
 	iowrite32be(tmp, tsi148_bridge->base + TSI148_LCSR_INTEO);
 
 	iowrite32be(TSI148_LCSR_INTC_LMC[monitor],
-		 tsi148_bridge->base + TSI148_LCSR_INTEO);
+		 tsi148_bridge->base + TSI148_LCSR_INTC);
 
 	/* Detach callback */
 	lm_callback[monitor] = NULL;
@@ -2582,13 +2581,6 @@ static void tsi148_remove(struct pci_dev
 	}
 
 	/*
-	 *  Disable and clear all interrupts.
-	 */
-	iowrite32be(0x0, tsi148_bridge->base + TSI148_LCSR_INTEO);
-	iowrite32be(0xFFFFFFFF, tsi148_bridge->base + TSI148_LCSR_INTC);
-	iowrite32be(0xFFFFFFFF, tsi148_bridge->base + TSI148_LCSR_INTEN);
-
-	/*
 	 *  Map all Interrupts to PCI INTA
 	 */
 	iowrite32be(0x0, tsi148_bridge->base + TSI148_LCSR_INTM1);
